### PR TITLE
Worked on Issue #20

### PR DESCRIPTION
**Pull request by Lukasaurus11**
- Refactored function get_most_starred_repos_python to get_most_starred_repos
- Updated requests parameters to include allow_redirects=False due to vulenabilities found https://devhub.checkmarx.com/cve-details/CVE-2023-32681/?utm_source=jetbrains&utm_medium=referral&utm_campaign=idea
- Updated .gitignore to include pycache

### CODE REVIEW

File: monkelib/API/get_data.py [hunk 1]
```diff
@@ -99,47 +102,52 @@ def get_user_repo(username: str) -> bool:
...
-    }
-    # Load the token from the pat.json file
+
+    queryList = []
+    for language in LANGUAGE_LIST:
+        queryList.append({
+            'q': f'language:{language}',
+            'per_page': 75,
+        })
+
```
Comment (felixggj) [inline monkelib/API/get_data.py:119]: What does this function do exactly? is the per_page: 75 the number of projects per language or what exactly?

File: monkelib/API/get_data.py [hunk 2]
```diff
@@ -99,47 +102,52 @@ def get_user_repo(username: str) -> bool:
...
+        except Exception as err:
+            print(f"An error occurred: {err}")
+            return False
+
+    dataList = [item for sublist in dataList for item in sublist]   # Flatten the list
+    _save_to_json(dataList, 'mostStarredRepos.json')
+    print("Data successfully written to mostStarredRepos.json")
+
+    _save_to_csv(dataList, 'mostStarredRepos.csv')
+    print("Data successfully written to mostStarredRepos.csv")
```
Comment (felixggj) [inline monkelib/API/get_data.py:152]: Why are we exporting to both json and to csv ? Pretty sure we just need csv

**Review by gabrieldeolaguibel — APPROVED** [review]
Looks good. Maybe in the future, for scaling, we can create multiple types of py scripts for a specific type of data we are retrieveing:
- Most starred repositories (general)
- User data
- etc

**Review by felixggj — APPROVED** [review]

File: monkelib/API/get_data.py (hunk 2 above)
Comment (gabrieldeolaguibel) [reply monkelib/API/get_data.py:152]: I think its good to store both. We can have the raw data and our transformed data

Comment (Lukasaurus11) [reply monkelib/API/get_data.py:152]: While I'm sure we are always going to work with just the CSV I was asked to save both at the start, and then we could remove this

**Comment by Lukasaurus11** [comment]
> Looks good. Maybe in the future, for scaling, we can create multiple types of py scripts for a specific type of data we are retrieveing:
> 
> * Most starred repositories (general)
> * User data
> * etc

Hey! As you can see this is already done, thanks our functional code approach. For example the function to get the users data is called get_user_repo, the one for most starred repos is get_most_starred_repos. We can expand to implement new functions in the future as new needs arise.

**Pull request merged**